server/src/test/scala/services/graphstages/ArrivalsGraphStageLiveBaseArrivalsSpec.scala

```diff
@@ -105,13 +106,176 @@ class ArrivalsGraphStageLiveBaseArrivalsSpec extends CrunchTestLike with AfterEa
     success
   }
 
+  "Given a live arrival with a scheduled arrival time and a base live arrival within 60 minutes scheduled arrival time" +
+    "Then they should get merged and have scheduled Departure" >> {
+    val probe = TestProbe("arrivals")
+    val (_, _, liveBaseSource, liveSource) = TestableArrivalsGraphStage(probe, buildArrivalsGraphStage).run
+
+    val scheduledDeparture = scheduled.addHours(-4).millisSinceEpoch
+    liveSource.offer(List(arrival()))
+
+    probe.fishForMessage(2 seconds) {
+      case ArrivalsDiff(toUpdate, _) =>
+        toUpdate.exists {
+          case (_, _) => true
+        }
+    }
+
+    liveBaseSource.offer(List(arrival(scheduledDate = scheduled.addMinutes(59).millisSinceEpoch, scheduledDepartureDate = Some(scheduledDeparture))))
+
+    probe.fishForMessage(2 seconds) {
```

Comment:
Seeing as we're offering an arrival and then checking for its existence 12 times in these tests, we could extract the process to a function (`offerArrivalAndCheck`?) which takes:
- the arrival
- the source to offer it to
- a partial function to check the existence of the arrival